### PR TITLE
Add ONNX configuration classes to docs

### PR DESCRIPTION
**Pull request by lewtun**
# What does this PR do?

This PR adds the ONNX configuration classes to the main doc. I've also removed the permalinks in the ONNX guide that we had included in #14904 to link to the configuration source code.

### CODE REVIEW

File: docs/source/_toctree.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@
   - local: debugging
     title: Debugging
   - local: serialization
-    title: Exporting transformers models
+    title: Exporting 🤗 Transformers models
```
Comment (lewtun) [inline docs/source/_toctree.yml:63]: I took the liberty of updating this title to match the convention used in the rest of the ToC

File: docs/source/serialization.mdx [hunk 2]
```diff
@@ -109,7 +109,7 @@ which should show the following logs:
 
 ```bash
 Validating ONNX model...
-        -[✓] ONNX model outputs' name match reference model ({'last_hidden_state'})
+        -[✓] ONNX model output names match reference model ({'last_hidden_state'})
```
Comment (lewtun) [inline docs/source/serialization.mdx:112]: This fixes a problem with syntax highlighting in the frontend


Screen Shot 2022-01-12 at 11 19 32

File: docs/source/serialization.mdx [hunk 3]
```diff
@@ -400,10 +398,8 @@ to the library, you will need to:
 
 * Implement the ONNX configuration in the corresponding `configuration_<model_name>.py`
 file
-* Include the model architecture and corresponding features in
-[`onnx.features.FeatureManager`](https://github.com/huggingface/transformers/blob/c4fa908fa98c3d538462c537d29b7613dd71306e/src/transformers/onnx/features.py#L71)
-* Add your model architecture to the tests in
-`test_onnx_v2.py`
+* Include the model architecture and corresponding features in [`~onnx.features.FeatureManager`]
```
Comment (sgugger) [inline docs/source/serialization.mdx:401]: This one is not documented so this link won't be resolved.

**Review by sgugger — APPROVED** [review]
Thanks a lot for adding this!

File: docs/source/serialization.mdx (hunk 3 above)
Comment (lewtun) [reply docs/source/serialization.mdx:401]: Good catch! Will add the doc for this class too :)

**Review by sgugger — COMMENTED** [review]
Oh, I just realized that you are using a path through modules, not the main init, so to link to `OnnxConfig`  for instance, you need to use the path [\`~onnx.config.OnnxConfig\`].

**Pull request merged**